modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientInboundMessageHandler.java

```diff
@@ -0,0 +1,557 @@
+
+    private static int getClientDataType(NativeTypeSpec spec) {
+        switch (spec) {
+            case INT8:
+                return ClientDataType.INT8;
+
+            case INT16:
+                return ClientDataType.INT16;
+
+            case INT32:
+                return ClientDataType.INT32;
+
+            case INT64:
+                return ClientDataType.INT64;
+
+            case FLOAT:
+                return ClientDataType.FLOAT;
+
+            case DOUBLE:
+                return ClientDataType.DOUBLE;
+
+            case DECIMAL:
+                return ClientDataType.DECIMAL;
+
+            case UUID:
+                return ClientDataType.UUID;
+
+            case STRING:
+                return ClientDataType.STRING;
+
+            case BYTES:
+                return ClientDataType.BYTES;
+
+            case BITMASK:
+                return ClientDataType.BITMASK;
+        }
+
+        throw new IgniteException("Unsupported native type: " + spec);
+    }
+
+    private void writeColumnValue(ClientMessagePacker packer, Tuple tuple, Column col) throws IOException {
+        var val = tuple.value(col.name());
+
+        if (val == null) {
+            packer.packNil();
+            return;
+        }
+
+        switch (col.type().spec()) {
+            case INT8:
+                packer.packByte((byte) val);
+                break;
+
+            case INT16:
+                packer.packShort((short) val);
+                break;
+
+            case INT32:
+                packer.packInt((int) val);
+                break;
+            case INT64:
+                packer.packLong((long) val);
+                break;
+
+            case FLOAT:
+                packer.packFloat((float) val);
+                break;
+
+            case DOUBLE:
+                packer.packDouble((double) val);
+                break;
+
+            case DECIMAL:
+                packer.packDecimal((BigDecimal) val);
+                break;
+
+            case UUID:
+                packer.packUuid((UUID) val);
+                break;
+
+            case STRING:
+                packer.packString((String) val);
+                break;
+
+            case BYTES:
+                byte[] bytes = (byte[]) val;
+                packer.packBinaryHeader(bytes.length);
+                packer.writePayload(bytes);
+                break;
+
+            case BITMASK:
+                packer.packBitSet((BitSet)val);
+                break;
+
+            default:
+                throw new IgniteException("Data type not supported: " + col.type());
+        }
+    }
+
+    private TableImpl createTable(ClientMessageUnpacker unpacker) throws IOException {
+        var size = unpacker.unpackMapHeader();
+        String name = null;
+        var settings = new Object() {
+            Integer replicas = null;
+            Integer partitions = null;
+            Map<String, Map<String, Object>> columns = null;
+        };
+
+        for (int i = 0; i < size; i++) {
```

Comment:
There were some plans to change current table configuration API, I'm still not sure which way this will go.

Let's handle it as a separate ticket, I'll remove this code for now.

https://issues.apache.org/jira/browse/IGNITE-15179